### DISCUSSION TITLE
Developing WatchOS apps with Titanium

### DISCUSSION THREAD

**Discussion by Dev-Adil** [appcelerator/titanium_mobile#13885] [General]
I'm looking to see if anyone has done this before. I've developed an app for iPhone/Android phones using Titanium SDKs, but now I want to create an app extension for Apple Watches. I've tried to use the information here, [Integrate a WatchKit App Built in Xcode,
](https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_How-tos/Platform_API_Deep_Dives/iOS_API_Deep_Dives/Integrate_a_WatchKit_App_Built_in_Xcode.html) but it doesn't dive much into the details or best practices. I'm particularly having trouble connecting the watch app with its parent app (via physical devices), and I can't seem to launch the watch app in visual studio code (where the parent app was developed). If anyone has done this before, would you care to share your experience and include some tips and best practices for those looking to also develop watch apps via Titanium?

**Comment by m1ga** [comment]
Hi, I only know this repo https://github.com/hansemannn/Titanium-WatchOS2-Sample-App but it is very old. Perhaps it has some useful information. Like is using https://titaniumsdk.com/api/titanium/watchsession.html to transfer data. 

But thats all I know, never used it or have any knowledge about the native side :smile:

**Reply by Dev-Adil** [reply]
Thanks Michael! I will check those links out and see what I can do with them. 

On another note: do you recommend developing the watch app on Xcode or VScode? I ask because the parent app does not seem to work properly if I try to build/run it on Xcode. But it runs perfectly fine on visual studio code when I build it using the Titanium extension.

**Reply by m1ga** [reply]
I don't do any native coding so I don't use Xcode. There is a workaround to build Ti apps with Xcode https://theappartisan.com/running-titanium-apps-from-xcode-a-workaround/ maybe that helps

oh, and there is https://github.com/tidev/titanium-sdk/pull/13032
